batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -181,15 +181,26 @@ public Integer fetchMaxId(String tableName) throws SQLException {
 					if (linkTemplate.containsKey("fhir") && linkTemplate.get("fhir") != null) {
 						String[] resourceName = linkTemplate.get("fhir").split("/");
 						if (resourceName.length >= 1 && resourceName[1].equals(search)) {
-							reverseMap.put(entry.getValue().getParentTable(), resourceName[1]);
+							if (!reverseMap.containsKey(resourceName[1])) {
+								reverseMap.put(resourceName[1], entry.getValue().getParentTable());
+							} else {
+								log.error("Some tables are mapped to the same Resources");
+								throw new Exception("config file has duplicate Resource mappings");
+							}
 						}
 					}
 				}
 			}
-			if (reverseMap.size() < searchList.length) {
-				log.error("Some of the passed FHIR resources are not mapped to any table, please check the config");
-			}
 			return reverseMap;
 		}
 	}
+	
+	public Map<String, ArrayList<String>> deduplicateReverseMap(Map<String, String> reverseMap) {
```

Comment:
I don't think we need to separate public methods for creation of reverse-map. Please expose only one of these and directly use that in FhirEtl.